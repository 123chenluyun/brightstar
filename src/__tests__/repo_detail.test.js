import React from 'react';
import ReactDOM from 'react-dom';
import RepoDetail from '../components/repo_detail.js';
import { mount,shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('<RepoDetail />', () => {

  let repo_detail, props;

  beforeEach(() => {
    props = {
      id:1, 
      details: [{avatar_url:"url1.com", url:"url1.com", login:"login1"},
                {avatar_url:"url2.com", url:"url2.com", login:"login2"},
                {avatar_url:"url3.com", url:"url3.com", login:"login3"}]

    };
    repo_detail = mount(<RepoDetail {...props}/> ); 
  });  

  it('renders correctly', () => {
    const repo_detail = renderer.create(
     <RepoDetail {...props}/>
    ).toJSON();
    expect(repo_detail).toMatchSnapshot();
  });

  it('allows us to set props', () => {
      expect(repo_detail.props()).toEqual(props);
  });

  it('RepoDetail has 3 contrib-cells', () => {
    expect(repo_detail.find('.contrib-link').length).toEqual(3);
    expect(repo_detail.find('.contrib-cell').length).toEqual(3);
  });

});