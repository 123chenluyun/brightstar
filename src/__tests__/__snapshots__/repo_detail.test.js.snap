// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoDetail /> renders correctly 1`] = `
<div>
  <div
    className="contrib-cell"
  >
    <img
      className="avatar"
      src="url1.com"
    />
    <a
      className="contrib-link"
      href="url1.com"
    >
      #1 
      login1
    </a>
  </div>
  <div
    className="contrib-cell"
  >
    <img
      className="avatar"
      src="url2.com"
    />
    <a
      className="contrib-link"
      href="url2.com"
    >
      #2 
      login2
    </a>
  </div>
  <div
    className="contrib-cell"
  >
    <img
      className="avatar"
      src="url3.com"
    />
    <a
      className="contrib-link"
      href="url3.com"
    >
      #3 
      login3
    </a>
  </div>
</div>
`;
